Port autoconf script to C99 and enable autoreconf for this package.

Upstrem bug tracked here:
https://bugzilla.netfilter.org/show_bug.cgi?id=1654

diff --git a/configure.ac b/configure.ac
index 060f307..fd70863 100644
--- a/configure.ac
+++ b/configure.ac
@@ -55,9 +55,9 @@ int main()
      struct in6_addr addr6;
      char buf[128];
      if (inet_ntop(AF_INET6, &addr6, buf, 128) == 0 && errno == EAFNOSUPPORT)
-        exit(1);
+        return 1;
      else
-        exit(0);
+        return 0;
   }
   ]])],[ AC_MSG_RESULT(yes)
        AC_DEFINE_UNQUOTED(HAVE_INET_NTOP_IPV6, 1, [Define to 1 if inet_ntop supports IPv6.])
